node.gyp

```diff
@@ -746,6 +748,47 @@
   ], # end targets
 
   'conditions': [
+    [ 'node_target_type=="static_library"', {
+      'targets': [
+        {
+          'target_name': 'static_node',
+          'type': 'executable',
+          'product_name': '<(node_core_target_name)',
+          'dependencies': [
+            '<(node_core_target_name)',
+          ],
+          'sources+': [
+            'src/node_main.cc',
+          ],
+          'include_dirs': [
+            'deps/v8/include',
+          ],
+          'xcode_settings': {
+            'OTHER_LDFLAGS': [
+              '-Wl,-force_load,<(PRODUCT_DIR)/libnode.a',
+            ],
+          },
+          'conditions': [
+            ['OS in "linux freebsd openbsd solaris android aix"', {
+              'ldflags': [
+                '-Wl,--whole-archive,'
+                '<(OBJ_DIR)/libnode.a '
+                '-Wl,--no-whole-archive',
+              ],
+            }],
+            ['OS=="win"', {
+              'msvs_settings': {
+                'VCLinkerTool': {
+                  'AdditionalOptions': [
+                    '/WHOLEARCHIVE:<(PRODUCT_DIR)/lib/node.lib'
```

Comment:
Could you define the three uses of the output lib path as `NODE_BASE`
probably something like:
```python
conditions : [[ 'OS=="win"', {
  variables: { NODE_OUTPUT_DIR: '<(PRODUCT_DIR)/lib' }
}, {
  variables: { NODE_OUTPUT_DIR: '<(PRODUCT_DIR)' }
]],
variables : {
  NODE_BASE: '>(NODE_OUTPUT_DIR)/<(STATIC_LIB_PREFIX)<(node_core_target_name)<(STATIC_LIB_SUFFIX)'
}
```

Reply:
Looks good, I'll take a look and update (might not be until after the weekend I'm afraid). Thanks